examples/partial_labels.py

```diff
@@ -0,0 +1,182 @@
+    )
+    return parser
+
+
+def main(
+    dev_host: str,
+    dev_api_token_path: str,
+    prod_host: str,
+    prod_api_token_path: str,
+    dev_dataset_id: int,
+    dev_questionnaire_id: int,
+    dev_model_group_id: int,
+    dev_workflow_id: int,
+    prod_workflow_id: int,
+    max_review_docs: int = 1000,
+    oversample_errs: bool = False,
+):
+    """
+    End to end function to data that has been submitted to a production workflow
+    and reviewed to an existing dataset, teach task, and extraction model
+
+    TODO
+     * Use max_review_docs and oversample_errs arguments
+
+    Args:
+        dev_host: Host URL for dev environment
+        dev_api_token_path: API token path for dev env
+        prod_host: Host URL for prod env
+        prod_api_token_path: API token path for prod env
+        dev_dataset_id: Dataset ID for dev dataset associated w/ prod workflow
+        dev_questionnaire_id: Questionnaire (teach task) ID associated with dev dataset.
+            Necessary because a dataset can have multiple teach tasks
+        dev_model_group_id: Model Group ID for dev dataset / teach task
+        dev_workflow_id: Workflow ID for dev workflow. Required for adding data
+            to teach task and model
+        prod_workflow_id: Workflow ID for prod workflow w/ review enabled
+        max_review_docs: Maximum number of documents from review to add to dataset
+            If max_review_docs is less than the number of docs that have been
+            reviewed, sample from review
+        oversample_errs: If True, sample more heavily from documents with more
+            discrepancies between model predictions and ground truth
+    """
+    # Connect to dev
+    dev_client = IndicoClient(
+        IndicoConfig(host=dev_host, api_token_path=dev_api_token_path)
+    )
+    dev_stagger = StaggeredLoop(client=dev_client)
+
+    # Get dataset details to determine when dataset was last updated. This will serve
+    # as a filter to prevent adding duplicate review data
+    dataset_details = dev_stagger.get_dataset_details(dataset_id=dev_dataset_id)
+
+    # Connect to prod
+    prod_client = IndicoClient(
+        IndicoConfig(host=prod_host, api_token_path=prod_api_token_path)
+    )
+    prod_stagger = StaggeredLoop(client=prod_client)
+
+    # Pull review data from prod workflow
+    workflow_results = prod_stagger.get_review_data(
+        workflow_id=prod_workflow_id,
+        update_date=datetime.fromtimestamp(float(dataset_details["updatedAt"])),
+    )
+
+    # StaggeredLoop class extends IndicoClient, so strictly has to connect to
+    # only one environment. As a result, we set the workflow_results attribute
+    # of the dev instance of the class using the review data pulled from prod.
+    dev_stagger.set_workflow_results(workflow_results)
+
+    # Get document bytes and text
+    dev_stagger.get_document_bytes()
+    dev_stagger.get_submission_full_text()
```

Comment:
We may not have a guarantee that the OCR text is identical after rerunning OCR, especially if the dataset settings are different or the PDF extraction version is different.